lib/ingreedy/root_parser.rb

```diff
@@ -87,7 +92,7 @@ class RootParser < Parslet::Parser
 
     rule(:reverse_format) do
       # e.g. flour 200g
-      ((whitespace >> quantity).absent? >> any).repeat.as(:ingredient) >>
+      ((whitespace >> quantity >> any.absent?).absent? >> any).repeat.as(:ingredient) >>
```

Comment:
Take for example `sugger 300g`: since `sugger` contains g in the quantity, it transitions to the next rule after reading to `sug' in continuous language.
Basically, nothing is considered to come after quantity, so `any.absent?`

Reply:
I don't quite follow this change, but i'm happy that if the tests are passing then it's probably fine 👍

Reply:
In continuation languages, where whitespace is treated as optional (denoted as "maybe"), parsing issues arise when encountering words that include units as substrings (for example, the letter "g" in "sugger"). This complication becomes apparent when tested locally.

The rule was designed to interpret strings structured as name+quantity. By ensuring no additional characters follow the quantity, we can eliminate the misinterpretation of substrings such as "g" in "suger" being wrongly recognized as a unit of quantity.

I hope I've explained it well.

Reply:
Oh great, I see! Thanks for explaining